tap_hubspot_beta/streams.py

```diff
@@ -2531,4 +2531,24 @@ class GeolocationSummaryMonthlyStream(FormsSummaryMonthlyStream):
         th.Property("breakdowns", th.CustomType({"type": ["array", "string"]})),
         th.Property("start_date", th.DateType),
         th.Property("end_date", th.DateType),
-    ).to_dict() 
+    ).to_dict()
+    
+def get_url_params_properties_if_too_long(params, fields):
+    def is_url_too_long(params):
+        return len(urlencode(params)) > 3000
+
+    properties_key = next((key for key in ["propertiesWithHistory", "properties"] if key in params), None)
+
+    if properties_key:
+        if isinstance(params[properties_key], list):
+            for chunk in params[properties_key]:
+                params_chunk = {**params, properties_key: chunk}
+                if is_url_too_long(params_chunk):
+                    params[properties_key] = fields
+                    return params
+        elif is_url_too_long(params):
+            params[properties_key] = fields
+    elif is_url_too_long(params):
+        params["properties"] = fields
+
+    return params
```

Comment:
i think you overcomplicated it here. if it was okay to have just the old set of fields, keep that logic. this code is hard to read/understand

Reply:
It didn't make sense to keep the old logic once we are doing the splitting for the API call. I can try to make it more readable though.

Reply:
@arilton done. Please let me know if it's better.